subsys/net/ip/net_pkt.c

```diff
@@ -1910,15 +1911,18 @@ int net_pkt_pull(struct net_pkt *pkt, size_t length)
 				pkt->buffer = buf->frags;
 				buf->frags = NULL;
 				net_buf_unref(buf);
-
-				net_pkt_cursor_init(pkt);
 			}
+
+			net_pkt_cursor_init(pkt);
```

Comment:
The `net_pkt_pull()` only pulls data from the beginning of the pkt, actually I wonder does it makes sense to restore the cursor at all.

Reply:
Clarification, the documentation says that the data is pulled from cursor, I probably missed something but did not quite understand how it could work by looking the original code.

Reply:
well again, forgot how this api works ... 
the doc say "Remove data from the packet at current location" so it removes what's in front of the cursor (I thought the opposite for some reason).

then no your patch is right, if the cursor was just initialized at the beginning of the pkt, but user request to remove <size of a net_buf>, then restoring is wrong indeed. 

So maybe it should not backup/restore, and always initialize at the end.

Reply:
> So maybe it should not backup/restore, and always initialize at the end.

Yes, it looks like the restoration can cause issues. I will experiment a bit about this.